core/src/stack/outputs.rs

```diff
@@ -40,20 +41,23 @@ impl StackOutputs {
             "overflow address outputs contain values that are not valid field elements",
         );
 
-        Self {
+        if stack.len() < STACK_TOP_SIZE {
+            return Err(OutputError::StackDepthTooSmall(stack.len(), STACK_TOP_SIZE));
+        }
+
+        // TODO: what validation should be applied to the overflow_addrs vector if stack.len() > 16?
```

Comment:
What's the advantage of returning an error from here rather than just padding with zeros?

Regarding `overflow_addrs` - yes, it would be good to check them too. Basically, number of values there should be equal to `stack.len() - 16`.

Also, if we are implementing proper error checking here, might be good to convert the `debug_assert!`'s above into errors as well.

Reply:
No advantage, happy to go with the padding route.